flink-ml-lib/src/main/java/org/apache/flink/ml/common/broadcast/operator/OneInputBroadcastWrapperOperator.java

```diff
@@ -0,0 +1,117 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.ml.common.broadcast.operator;
+
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.ml.iteration.datacache.nonkeyed.DataCacheReader;
+import org.apache.flink.streaming.api.operators.OneInputStreamOperator;
+import org.apache.flink.streaming.api.operators.StreamOperatorFactory;
+import org.apache.flink.streaming.api.operators.StreamOperatorParameters;
+import org.apache.flink.streaming.api.watermark.Watermark;
+import org.apache.flink.streaming.runtime.streamrecord.LatencyMarker;
+import org.apache.flink.streaming.runtime.streamrecord.StreamRecord;
+import org.apache.flink.streaming.runtime.watermarkstatus.WatermarkStatus;
+
+/** Wrapper for WithBroadcastOneInputStreamOperator. */
+public class OneInputBroadcastWrapperOperator<IN, OUT>
+        extends AbstractBroadcastWrapperOperator<OUT, OneInputStreamOperator<IN, OUT>>
+        implements OneInputStreamOperator<IN, OUT> {
+
+    public OneInputBroadcastWrapperOperator(
+            StreamOperatorParameters<OUT> parameters,
+            StreamOperatorFactory<OUT> operatorFactory,
+            String[] broadcastStreamNames,
+            TypeInformation[] inTypes,
+            boolean[] isBlocking) {
+        super(parameters, operatorFactory, broadcastStreamNames, inTypes, isBlocking);
+    }
+
+    @Override
+    public void processElement(StreamRecord<IN> streamRecord) throws Exception {
+        if (isBlocking[0]) {
+            if (areBroadcastVariablesReady()) {
+                dataCacheWriters[0].finishCurrentSegmentAndStartNewSegment();
+                segmentLists[0].addAll(dataCacheWriters[0].getNewlyFinishedSegments());
+                if (segmentLists[0].size() != 0) {
+                    DataCacheReader dataCacheReader =
+                            new DataCacheReader<>(
+                                    inTypes[0].createSerializer(
+                                            containingTask.getExecutionConfig()),
+                                    fileSystem,
+                                    segmentLists[0]);
+                    while (dataCacheReader.hasNext()) {
+                        wrappedOperator.processElement(new StreamRecord(dataCacheReader.next()));
+                    }
+                }
+                segmentLists[0].clear();
+                wrappedOperator.processElement(streamRecord);
+
+            } else {
+                dataCacheWriters[0].addRecord(streamRecord.getValue());
+            }
+
+        } else {
+            while (!areBroadcastVariablesReady()) {
+                mailboxExecutor.yield();
+            }
+            wrappedOperator.processElement(streamRecord);
+        }
+    }
```

Comment:
Perhaps we could extract a common method in the abstract base class ? 

```
protected void xxx(int inputIndex, ThrowingConsumer consumer) {
      ...  
}
```

Then we could avoid repeat logic in all the processElementX by calling `xxxx(0, this::processElement)`.

Reply:
Thanks Yun. I have finished the refactoring, please refer to `AbstractBroadcastWrapperOperator#processElementX()` and `AbstractBroadcastWrapperOperator#endInputX()`